src/credentialstore/credentialstore.ts

```diff
@@ -26,44 +33,50 @@ export class CredentialStore implements ICredentialStore {
 	 * @param {string} credentialId the ID uniquely identifying this credential
 	 * @returns {Promise<Credential>} Promise that resolved to the credential, or undefined if not found
 	 */
-	public readCredential(credentialId: string): Promise<Contracts.Credential> {
+	public async readCredential(credentialId: string): Promise<Contracts.Credential> {
 		let self = this;
 		let cred: Contracts.Credential = new Contracts.Credential();
 		cred.credentialId = credentialId;
-		return new Promise<Contracts.Credential>((resolve, reject) => {
-			self._client
-				.sendRequest(Contracts.ReadCredentialRequest.type, cred)
-				.then(returnedCred => {
-					resolve(returnedCred);
-				}, err => reject(err));
-		});
+		try {
+			const returnedCred = await self._client.sendRequest(Contracts.ReadCredentialRequest.type, cred)
+			return returnedCred;
+		} catch (error) {
+			throw error;
+		}
 	}
 
-
-	public saveCredential(credentialId: string, password: any): Promise<boolean> {
+	public async saveCredential(credentialId: string, password: any): Promise<boolean> {
 		let self = this;
 		let cred: Contracts.Credential = new Contracts.Credential();
 		cred.credentialId = credentialId;
 		cred.password = password;
-		return new Promise<boolean>((resolve, reject) => {
-			self._client
-				.sendRequest(Contracts.SaveCredentialRequest.type, cred)
-				.then(status => {
-					resolve(status);
-				}, err => reject(err));
-		});
+		/* This is only done for linux because this is going to be
+		* the default credential system for linux in the next release
```

Comment:
I would avoid using terms like "next release" in comments since that relies on the person having context of when "next" is.